### ISSUE TITLE
[MLForecast: test set eval and early stopping ] 

### ISSUE THREAD

**Issue by Djuseppe** [Nixtla/mlforecast#309]
### Description

Hey team,
it's gonna be beneficial to have an option to early stop models when training. 

### Use case

CV hyper-param optimization

**Comment by jmoralez** [comment]
Hey @Djuseppe, thanks for using mlforecast. We have that built-in for LightGBM in the [LightGBMCV class](https://nixtlaverse.nixtla.io/mlforecast/lgb_cv.html) and you can do it for other frameworks by using [custom training](https://nixtlaverse.nixtla.io/mlforecast/docs/how-to-guides/custom_training.html#custom-training). It'd be very hard to have a general solution because each framework has a slightly different interface.

**Comment by Djuseppe** [comment]
thanks for the info 🙂 
indeed a custom training is amazing and suits me nicely, since I have also custom CV with Optuna optimization.
Closing, as existing features are just enough.
